apps/04TowersOfHanoi.js

```diff
@@ -16,30 +16,51 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
     // Your code here
+    stacks[endStack].push(stacks[startStack].pop());
 
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
     // Your code here
+    var arrayStart = stacks[startStack];
+    
+    var arrayEnd = stacks[endStack];
+    
+    if (arrayStart[arrayStart.length -1] < arrayEnd[arrayEnd.length -1] || arrayEnd.length === 0) {
+        return true;
+    }
+    else {
+        return false;
+    };
 
 }
 
 function checkForWin() {
     // Your code here
+    if (stacks.b.length === 4 || stacks.c.length === 4) {
+        console.log("You Win!");
+        return true;
+    } else {
+        return false;
+    };
 
 }
 
 function towersOfHanoi(startStack, endStack) {
-    // Your code here
+   isLegal(startStack, endStack);
```

Comment:
While you are calling isLegal here, you don't have any code to use the result of the function to control whether or not to move the piece. The correct approach:

if(isLegal(startStack, endStack){
//move piece
//check for win
}